File: OrderAPI/Models/DTOs/OrderRequestDTO.cs
using Swashbuckle.AspNetCore.Annotations;

namespace OrderAPI.Models.DTOs
{
    public class OrderRequestDTO
    {
        [SwaggerSchema("Order's unique identifier", ReadOnly = true)]
        public Guid Id { get; set; }

        [SwaggerSchema("Timestamp when the order was created", ReadOnly = true)]
        public DateTime CreatedAt { get; set; }

        [SwaggerSchema("Timestamp when the order was last updated", ReadOnly = true)]
        public DateTime? UpdatedAt { get; set; }

        [SwaggerSchema("Timestamp when the order was deleted", ReadOnly = false)]
        public Guid CustomerId { get; set; }

        [SwaggerSchema("Current status of the order")]
        public OrderStatus Status { get; set; }

        [SwaggerSchema("List of items in the order")]
        public required List<OrderItemDTO> Items { get; set; }
    }
}

File: models/dtos/order_request_dto.py
import uuid
from dataclasses import dataclass, field
from datetime import datetime
from typing import List, Optional
from models.dtos.order_item_dto import OrderItemDTO
from models.entities.order_status import OrderStatus


@dataclass
class OrderRequestDTO:

    id: Optional[uuid.UUID] = None

    created_at: Optional[datetime] = None

    updated_at: Optional[datetime] = None

    customer_id: uuid.UUID = field(default_factory=uuid.uuid4)

    status: Optional[OrderStatus] = None

    items: List[OrderItemDTO] = field(default_factory=list)
